### PR TITLE
Generate aten_interned_strings.h automatically

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #69407

This generates aten_interned_strings.h from `native_functions.yaml`
which is more like how it was originally done. The items deleted from
`interned_strings.h` are duplicates that need to be removed in order
for the code to compile, some of the remaining items may still be out
of date but it is fairly benign even if that's the case.

Differential Revision: [D32923636](https://our.internmc.facebook.com/intern/diff/D32923636)

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -162,6 +162,7 @@ genrule(
         "aten/src/ATen/MetaFunctions.h",
         "aten/src/ATen/MetaFunctions_inl.h",
         "aten/src/ATen/NativeMetaFunctions.h",
+        "aten/src/ATen/core/aten_interned_strings.h",
```
Comment (albanD) [inline BUILD.bazel:165]: Why does this need to be added? The file was there before no?

Comment (peterbell10) [reply BUILD.bazel:165]: bazel/buck needs to know all the outputs of generated code or it ignores them. So you'd get missing include errors like this:
https://github.com/pytorch/pytorch/runs/4414879421?check_suite_focus=true#step:16:25519

Comment (albanD) [reply BUILD.bazel:165]: Ok! That's what I was expecting. Thanks for the confirmation.

**Review by albanD — APPROVED** [review]

**Comment by albanD** [comment]
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32923636).

**Comment by albanD** [comment]
Hey @peterbell10 sorry this dropped out of my queue :( 
Could you rebase on top of master please and I'll land

**Comment by peterbell10** [comment]
@albanD rebased on master

**Review by albanD — APPROVED** [review]

**Comment by albanD** [comment]
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32923636).

**Comment by albanD** [comment]
I'm not sure why I failed to land this again :( Sorry! It is doing merge conflicts again.

**Comment by peterbell10** [comment]
@albanD fixed  merge conflicts and CI failures so far look unrelated.

**Comment by albanD** [comment]
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32923636).

**Comment by peterbell10** [comment]
Rebased just to let CI run.

**Comment by albanD** [comment]
It is already trying to land. So leaving it as is.
Will re-import if the landing fails again.

**Comment by albanD** [comment]
It did went through :) 
Thanks again Peter for fixing this!

**Pull request closed without merging**